### PR TITLE
Cleanup: improve the use of maps

### PR DESCRIPTION
**Pull request by yeedove**
**Please provide a description of this PR:**

improve the use of maps

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -350,12 +352,8 @@ func (c *Controller) MCSServices() []model.MCSServiceInfo {
...
 	}
 
-	out := make([]model.MCSServiceInfo, 0, len(outMap))
-	for _, v := range outMap {
-		out = append(out, *v)
-	}
-
-	return out
+	out := maps.Values(outMap)
+	return slices.Dereference(out)
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:356]: why not return out

Comment (yeedove) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:356]: should return []*model.MCSServiceInfo) not []model.MCSServiceInfo, slices.Dereference() can do it

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:356]: can we change outMap to []model.MCSServiceInfo, then we donot need to convert any more

Comment (yeedove) [reply pilot/pkg/serviceregistry/kube/controller/controller.go]: Thank review, I simplified it with your suggestion

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 2]
```diff
@@ -350,12 +349,7 @@ func (c *Controller) MCSServices() []model.MCSServiceInfo {
 		mcsService.ClusterSetVIP = si.clusterSetVIP
 	}
 
-	out := make([]model.MCSServiceInfo, 0, len(outMap))
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:353]: I think this was intentionally a pointer so the import and export loops would modify the same thing. We lost that.

**Review by howardjohn — COMMENTED** [review]
Mostly LGTM

**Review by howardjohn — APPROVED** [review]

**Pull request merged**